cmd/resolutioncli/catalogmetadata_reader.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	catalogd "github.com/operator-framework/catalogd/api/core/v1alpha1"
+	"github.com/operator-framework/operator-registry/alpha/action"
+	"github.com/operator-framework/operator-registry/alpha/declcfg"
+)
+
+type catalogmetadataReader struct {
+	renderer      action.Render
+	clientBuilder *fake.ClientBuilder
```

Comment:
Why is this using fakes?

Reply:
Resolution CLI is a PoC for offline resolution. It doesn't interact with cluster. I used fakes here as a quick way to keep the rest of the code untouched, but now given that we will be removing kube client in #411 I think we should just create a second implementation of a Client which reads from a FS and has the same interfacte as the `Client` from `internal/catalogmetadata/client`.

Should save us some marshalling too.

Reply:
Replaced with a client.